crypto/src/proofs/transparent.rs

```diff
@@ -122,7 +122,7 @@ impl OutputProof {
         gadgets::note_commitment_integrity(self.note.clone(), note_commitment)?;
 
         gadgets::balance_commitment_integrity(
-            -balance_commitment,
+            balance_commitment,
```

Comment:
@redshiftzero is it actually correct to remove the minus here?

`OutputProof::verify` expects a balance commitment that's negative (or, does it?), and the gadget will check integrity against the note commitment (positive, afaict). I could make `test_output_proof_verification_success` pass by feeding `verify` with a positive balance commitment, but since it's an `OutputProof` shouldn't it be negative?

Reply:
good question: the output proof doesn't check the signs, it's only checking that the commitment has been formed properly, i.e. that the value (should always be positive even for something that will ultimately subtract from the value balance in the transaction) and the blinding factor are an opening of the public value/balance commitment. Later on when we sum the commitments (the verifier does this) the appropriate sign will be used to ensure that the action has the appropriate contribution to the value balance, and that is checked cryptographically using the binding signature at the transaction level.

Reply:
got it, but the balance commitment won't open to the note value commitment because they're different:

the balance commitment:

$cv_o = [-v_o] G_v + [rcv_o] G_{rcv}$

the note commitment, in the happy path:

$cv_o = [v_o] G_v + [rcv_o] G_{rcv}$

This is how $cv_o$ is computed in `OutputPlan::output_body`:
```rust
     let balance_commitment = self.balance().commit(self.value_blinding); // balance is negated
```

on the other hand, this works well for `SpendProof`, so maybe we could modify `gadgets::balance_commitment_integrity` to take balances instead of values:

```rust
/// Check the integrity of the balance (previously value) commitment.
pub(crate) fn balance_commitment_integrity(
    balance_commitment: balance::Commitment,
    balance_blinding: Fr,
    balance: Balance,
) -> Result<()> {
    if balance_commitment != balance.commit(value_blinding) {
        Err(anyhow!("balance commitment mismatch"))
    } else {
        Ok(())
    }
}
```

where `balance` is built from the note value, and `blinding` is specified within the proof. In the case of `OutputProof` we would negate the balance, and in the case of `SpendProof` we wouldn't. This way we get closer to eliminating `Value::commit` and the gadget  logic doesn't change.

I think I would also add a couple tests in `OutputPlan` and `SpendPlan` to anchor the way we prepare / verify commitments, since `penumbra_transaction` imports `penumbra_crypto`.

Reply:
ahhh gotcha, so since the change in your PR here now uses `Balance` in `OutputPlan::output_body`, we'd also need to flip the sign of the blinding factor for all else to stay the same. Instead I think your suggestion to modify the balance commitment integrity check to take a balance could be good. Tests to anchor this behavior in penumbra_transaction would also be great!